google-cloud-core/src/test/java/com/google/cloud/ServiceOptionsTest.java

```diff
@@ -313,4 +314,16 @@ public void testGetServiceAccountProjectId_nonExistentFile() throws Exception {
 
     assertNull(ServiceOptions.getServiceAccountProjectId(credentialsFile.getPath()));
   }
+
+  @Test
+  public void testValidateProjectId() throws Exception {
+    String validProjectId = "abc-123";
+    String invalidProjectId1 = "abc=123";
+    String invalidProjectId2 = "abc123-";
+    String invalidProjectId3 = "1abc-23";
+    assertThat(ServiceOptions.isValidProjectId(validProjectId)).isTrue();
+    assertThat(ServiceOptions.isValidProjectId(invalidProjectId1)).isFalse();
+    assertThat(ServiceOptions.isValidProjectId(invalidProjectId2)).isFalse();
+    assertThat(ServiceOptions.isValidProjectId(invalidProjectId3)).isFalse();
```

Comment:
I think you can just do `assertThat(ServiceOptions.isaValidProjectId("1abc-23")).isFalse()`. No need for indirection.